src/functions/template-tags/general.php

```diff
@@ -1222,12 +1222,17 @@ function tribe_events_template_data( $event, array $additional = null ) {
 			$image_tool_src = $image_tool_arr[0];
 		}
 
-		if ( has_excerpt( $event->ID ) ) {
+		if ( post_password_required( $event->ID ) ) {
+			$password_required_msg = __( 'You must visit this event and enter the password to view the description.', 'tribe-events-calendar' );
+			$excerpt = apply_filters( 'tribe_events_template_data_password_required', $password_required_msg );
+			$do_not_truncate = true;
+		}
+		elseif ( has_excerpt( $event->ID ) ) {
 			$excerpt = $event->post_excerpt;
 		} else {
 			$excerpt = $event->post_content;
 		}
-		$excerpt = Tribe__Events__Main::instance()->truncate( $excerpt, 30 );
+		$excerpt = isset( $do_not_truncate ) ? $excerpt : Tribe__Events__Main::instance()->truncate( $excerpt, 30 );
```

Comment:
Maybe check the truthiness of `$do_not_truncate` at the same time with `empty()` rather than `isset()`?